src/data/fintype/card.lean

```diff
@@ -184,6 +206,38 @@ begin
   { simp }
 end
 
+@[to_additive]
+lemma fintype.prod_fiberwise [fintype α] [fintype β] [decidable_eq β] [comm_monoid γ]
+  (f : α → β) (g : α → γ) :
+  (∏ b : β, ∏ a : {a // f a = b}, g (a : α)) = ∏ a, g a :=
+begin
+  rw [← finset.prod_equiv (equiv.sigma_preimage_equiv f) _,
+      ← finset.univ_sigma_univ, finset.prod_sigma],
+  refl
+end
+
+section
+open finset
+
+variables {α₁ : Type*} {α₂ : Type*} {M : Type*} [fintype α₁] [fintype α₂] [comm_monoid M]
+
+@[to_additive]
+lemma fintype.prod_sum_type (f : α₁ ⊕ α₂ → M) :
+  (∏ x, f x) = (∏ a₁, f (sum.inl a₁)) * (∏ a₂, f (sum.inr a₂)) :=
```

Comment:
This'll come in handy. I realised that `sum_bind` is pretty unpleasant to use and `sum_sigma` should always be used instead. The same is true of `sum_union`, but we don't have a version of `finset.sigma` for `sum` types, so I thought I'd define one. This is all irrelevant to this PR>

Reply:
Ok, glad you like it.